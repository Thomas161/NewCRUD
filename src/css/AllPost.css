.divElement {
  align-content: center;
  /* margin: 70px; */
}
.divElement h1 {
  font-family: tahoma;
  font-style: italic;
  letter-spacing: 1em;
  border: 4px gold solid;
  box-shadow: 0 8px 8px 8px gray;
  margin: 70px;
  width: 550px;
  color: #f3be18;
  padding: 2%;
  background-color: #ff937b;
}
.divElement input {
  margin: 70px;
  padding: 10px 10px 10px 10px;
  width: 30%;
  border-radius: 20px;
  color: white;
  background-color: gray;
}
.divElement button {
  margin: 70px;
  padding: 10px 10px 10px 10px;
}
