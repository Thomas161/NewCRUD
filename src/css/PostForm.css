h1 {
  color: black;
  text-shadow: 1px 1px;
  padding: 3%;
}
input {
  padding: 10px 10px 10px 10px;
  width: 40%;
  border-radius: 15px;
}
#divElement {
  position: relative;
  align-content: center;
  top: 40%;
  left: 40%;
  margin-top: -30px;
  margin-left: -30px;
  width: 80%;
  height: 80%;
}
