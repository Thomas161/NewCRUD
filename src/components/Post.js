import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { EDIT_POST, DELETE_POST } from "../actions";
import { CSSTransition } from "react-transition-group";
import s from "../css/Post.module.css";
import "../css/style.css";

class Post extends Component {
  render() {
    const {
      post: { title, message }
    } = this.props;
    const { head } = this.props;
    return (
      <div className={s.pos}>
        <div>
          <h1 className={s.h1}>{head}</h1>
        </div>
        <CSSTransition classNames="list" timeout={1500}>
          <div className="ui card">
            <div className="content">
              <h3 className={s.h3}>{title}</h3>
              <h4 className={s.h4}>{message}</h4>

              <button
                onClick={() =>
                  this.props.dispatch({
                    type: EDIT_POST,
                    id: this.props.post.id
                  })
                }
                className="circular ui icon button green"
              >
                <i className="icon edit"></i>
                Edit
              </button>
              <button
                onClick={() =>
                  this.props.dispatch({
                    type: DELETE_POST,
                    id: this.props.post.id
                  })
                }
                className="circular ui icon button red"
              >
                <i className="icon trash"></i>
                Delete
              </button>
            </div>
          </div>
        </CSSTransition>
      </div>
    );
  }
}
Post.propTypes = {
  head: PropTypes.string.isRequired
};
Post.defaultProps = {
  head: "All Posts"
};

export default connect()(Post);
