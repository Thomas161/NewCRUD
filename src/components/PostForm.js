import React, { Component } from "react";
import { connect } from "react-redux";
import "../css/PostForm.css";
import uuid from "uuid";
class PostForm extends Component {
  handleSubmit = e => {
    e.preventDefault();
    const title = this.getTitle.value;
    const message = this.getMessage.value;
    const data = {
      id: uuid.v4(),
      title,
      message,
      editing: false
    };

    console.log(data);

    this.props.dispatch({
      type: "ADD_POST",
      data
    });
    this.getTitle.value = "";
    this.getMessage.value = "";
  };
  componentDidMount() {
    console.log("Mounted");
  }

  render() {
    return (
      <div id="divElement">
        <h1>Create Post</h1>
        <form className="pure-form" onSubmit={this.handleSubmit}>
          <input
            required
            type="text"
            ref={input => (this.getTitle = input)}
            placeholder="Title"
          />
          <br />

          <input
            type="text"
            ref={input => (this.getMessage = input)}
            placeholder="Message"
          />
          <br />
          <br />

          <button className="ui inverted violet button">Post</button>
        </form>
      </div>
    );
  }
}
export default connect()(PostForm);
