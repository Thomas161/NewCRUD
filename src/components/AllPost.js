import React, { Component } from "react";
import { connect } from "react-redux";
import Post from "./Post";
import EditComponent from "./EditComponent";
import "../css/AllPost.css";
import PropTypes from "prop-types";
class AllPost extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div id="h1">
        <h1>{this.props.head}</h1>
        {this.props.posts.map(post => (
          <div key={post.id}>
            {post.editing ? (
              <EditComponent post={post} key={post.id} />
            ) : (
              <Post key={post.id} post={post} />
            )}
          </div>
        ))}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    posts: state
  };
};

AllPost.propTypes = {
  head: PropTypes.string.isRequired
};
AllPost.defaultProps = {
  head: "All Posts"
};

export default connect(mapStateToProps)(AllPost);
