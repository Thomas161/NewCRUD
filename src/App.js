import React, { Component } from "react";
import PostForm from "../src/components/PostForm";
import AllPost from "../src/components/AllPost";
class App extends Component {
  render() {
    return (
      <div
        className="App"
        style={{
          backgroundColor: "orange",
          width: "100%",
          height: "100%"
        }}
      >
        <PostForm />
        <hr />
        <AllPost />
      </div>
    );
  }
}

export default App;
