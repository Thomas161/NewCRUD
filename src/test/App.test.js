import React from "react";
import App from "../App";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import sinon from "sinon";
import AllPost from "../components/AllPost";
import Post from "../components/Post";
import PostForm from "../components/PostForm";
Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureMockStore();
const store = mockStore({});

describe("Renders Component", () => {
  it("Renders Parent that holds store", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(AllPost).toBeDefined();
    expect(PostForm).toBeDefined();
  });
});
describe("renders child component", () => {
  it("<Allpost/>", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <AllPost />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Component has button", () => {
  it("<Post/> has a button", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Post />
      </Provider>
    );

    expect(wrapper.contains("<button></button>")).toBe(false);
  });
});

describe("checking objects in child", () => {
  it("<PostForm/> contains data{}", () => {
    const data = {
      editing: false
    };
    const wrapper = shallow(
      <Provider store={store}>
        <PostForm />
      </Provider>
    );
    expect(wrapper.update());
    expect(data).toHaveProperty("editing");
  });
});
