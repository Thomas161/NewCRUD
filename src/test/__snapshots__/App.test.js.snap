// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Component Renders Parent that holds store 1`] = `ShallowWrapper {}`;

exports[`renders child component <Allpost/> 1`] = `ShallowWrapper {}`;
